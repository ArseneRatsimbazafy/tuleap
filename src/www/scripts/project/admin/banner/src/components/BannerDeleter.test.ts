/*
 * Copyright (c) Enalean, 2019 - Present. All Rights Reserved.
 *
 *  This file is a part of Tuleap.
 *
 *  Tuleap is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  Tuleap is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import BannerDeleter from "./BannerDeleter.vue";
import { createProjectAdminBannerLocalVue } from "../helpers/local-vue-for-tests";

describe("BannerDeleter", () => {
    it("displays a delete button", async () => {
        const wrapper = shallowMount(BannerDeleter, {
            localVue: await createProjectAdminBannerLocalVue()
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("emits a delete delete-banner event on click", async () => {
        const wrapper = shallowMount(BannerDeleter, {
            localVue: await createProjectAdminBannerLocalVue()
        });

        const emitSpy = jest.spyOn(wrapper.vm, "$emit");

        wrapper.find("button").trigger("click");
        expect(emitSpy).toBeCalledWith("delete-banner");
    });

    it("disables the button and displays a transition test while requesting deletion", async () => {
        const wrapper = shallowMount(BannerDeleter, {
            localVue: await createProjectAdminBannerLocalVue()
        });

        wrapper.find("button").trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });
});
