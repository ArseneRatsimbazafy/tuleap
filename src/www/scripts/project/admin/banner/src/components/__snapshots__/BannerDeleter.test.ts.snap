// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerDeleter disables the button and displays a transition test while requesting deletion 1`] = `
<button
  class="tlp-button-danger"
  disabled="disabled"
  type="button"
>
  <i
    class="tlp-button-icon fa fa-fw fa-spin fa-circle-o-notch"
  />
   
  <span
    data-current-language="en_US"
    data-msgid="Remove banner"
  >
    Remove banner
  </span>
</button>
`;

exports[`BannerDeleter displays a delete button 1`] = `
<button
  class="tlp-button-danger"
  type="button"
>
  <!---->
   
  <span
    data-current-language="en_US"
    data-msgid="Remove banner"
  >
    Remove banner
  </span>
</button>
`;
